test/main.hs

```diff
@@ -81,6 +81,17 @@ main = do
       unless (e1 == ExitSuccess && e2 == ExitSuccess)
             $ error "sleep exited with non-zero exit code!"
 
+    do
```

Comment:
The test implementation is pretty fragile.

* The sleep value is low enough that the process could legitimately finish before the `waitForProcess` begins
* I'd rather not rely on `BlockedIndefinitelyOnMVar`, but instead use `onException` to `putMVar` a value in the case of an exception
* It's worth reporting the `ec` in the `Right` case; an exit code of 0 vs 127 will be highly informative, for instance